storage/backup/backup-cli/src/backup_types/transaction/backup.rs

```diff
@@ -0,0 +1,197 @@
+    ) -> Self {
+        Self {
+            start_version: opt.start_version,
+            num_transactions: opt.num_transactions,
+            max_chunk_size: global_opt.max_chunk_size,
+            client,
+            storage,
+        }
+    }
+
+    pub async fn run(self) -> Result<FileHandle> {
+        let backup_handle = self.storage.create_backup(&self.backup_name()).await?;
+
+        let mut chunks = Vec::new();
+        let mut chunk_bytes = Vec::new();
+
+        let mut transactions_file = self
+            .client
+            .get_transactions(self.start_version, self.num_transactions)
+            .await?;
+        let mut current_ver: u64 = self.start_version;
+        let mut chunk_first_ver: u64 = self.start_version;
+
+        while let Some(record_bytes) = transactions_file.read_record_bytes().await? {
+            if chunk_bytes.len() + size_of::<u32>() + record_bytes.len() > self.max_chunk_size {
+                ensure!(
+                    !chunk_bytes.is_empty(),
+                    "max chunk size too small: {}",
+                    self.max_chunk_size
+                );
+                assert!(chunk_bytes.len() <= self.max_chunk_size);
+                println!("Reached max_chunk_size.");
+
+                let chunk = self
+                    .write_chunk(
+                        &backup_handle,
+                        &chunk_bytes,
+                        chunk_first_ver,
+                        current_ver - 1,
+                    )
+                    .await?;
+                chunks.push(chunk);
+                chunk_bytes = vec![];
+                chunk_first_ver = current_ver;
+            }
+
+            chunk_bytes.extend(&(record_bytes.len() as u32).to_be_bytes());
+            chunk_bytes.extend(&record_bytes);
+            current_ver += 1;
+        }
+
+        assert!(!chunk_bytes.is_empty());
+        assert!(chunk_bytes.len() <= self.max_chunk_size);
+        assert_eq!(
+            current_ver,
+            self.start_version + self.num_transactions as u64
+        );
+        println!("Last chunk.");
+        let chunk = self
+            .write_chunk(
+                &backup_handle,
+                &chunk_bytes,
+                chunk_first_ver,
+                current_ver - 1,
+            )
+            .await?;
+        chunks.push(chunk);
+
+        self.write_manifest(&backup_handle, self.start_version, current_ver - 1, chunks)
+            .await
+    }
+}
+
+impl TransactionBackupController {
+    fn backup_name(&self) -> ShellSafeName {
+        format!("transaction_{}-", self.start_version)
+            .try_into()
+            .unwrap()
+    }
+
+    fn manifest_name() -> &'static ShellSafeName {
+        static NAME: Lazy<ShellSafeName> =
+            Lazy::new(|| ShellSafeName::from_str("transaction.manifest").unwrap());
+        &NAME
+    }
+
+    fn chunk_name(first_ver: Version) -> ShellSafeName {
+        format!("{}-.chunk", first_ver).try_into().unwrap()
```

Comment:
why do we naming it `-` ?

Reply:
it's a chunk containing versions A to B, while B MAY not be known yet when we open the file. (And it follows the same convention other backup types use.)

Reply:
sort of ugly though but I don't mind so badly.